cmd/src/gateway_benchmark.go

```diff
@@ -0,0 +1,258 @@
+			fmt.Printf("\nTesting %s...", name)
+
+			for i := 0; i < *requestCount; i++ {
+				duration := benchmarkEndpoint(client, url)
+				if duration > 0 {
+					durations = append(durations, duration)
+				}
+				fmt.Printf("\rTesting %s: %d/%d", name, i+1, *requestCount)
+			}
+			fmt.Println()
+
+			avg, p5, p95, median, total := calculateStats(durations)
+
+			results = append(results, endpointResult{
+				name:       name,
+				avg:        avg,
+				median:     median,
+				p5:         p5,
+				p95:        p95,
+				total:      total,
+				successful: len(durations),
+			})
+		}
+
+		printResults(results, requestCount)
+		return nil
+	}
+
+	gatewayCommands = append(gatewayCommands, &command{
+		flagSet: flagSet,
+		aliases: []string{},
+		handler: handler,
+		usageFunc: func() {
+			_, err := fmt.Fprintf(flag.CommandLine.Output(), "Usage of 'src gateway %s':\n", flagSet.Name())
+			if err != nil {
+				return
+			}
+			flagSet.PrintDefaults()
+			fmt.Println(usage)
+		},
+	})
+}
+
+type endpointResult struct {
+	name       string
+	avg        time.Duration
+	median     time.Duration
+	p5         time.Duration
+	p95        time.Duration
+	total      time.Duration
+	successful int
+}
+
+func benchmarkEndpoint(client *http.Client, url string) time.Duration {
+	start := time.Now()
+	resp, err := client.Get(url)
+	if err != nil {
+		fmt.Printf("Error calling %s: %v\n", url, err)
+		return 0
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Printf("Error closing response body: %v\n", err)
+		}
+	}(resp.Body)
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		fmt.Printf("Error reading response body: %v\n", err)
+		return 0
+	}
+	fmt.Printf("Response from %s: %s\n", url, body)
+
+	return time.Since(start)
+}
+
+func calculateStats(durations []time.Duration) (time.Duration, time.Duration, time.Duration, time.Duration, time.Duration) {
+	if len(durations) == 0 {
+		return 0, 0, 0, 0, 0
+	}
+
+	sort.Slice(durations, func(i, j int) bool {
+		return durations[i] < durations[j]
+	})
+
+	var sum time.Duration
+	for _, d := range durations {
+		sum += d
+	}
+	avg := sum / time.Duration(len(durations))
+
+	p5idx := int(float64(len(durations)) * 0.05)
+	p95idx := int(float64(len(durations)) * 0.95)
+	medianIdx := len(durations) / 2
+
+	return avg, durations[p5idx], durations[p95idx], durations[medianIdx], sum
+}
+
+func formatDuration(d time.Duration, best bool, worst bool) string {
+	value := fmt.Sprintf("%.2fms", float64(d.Microseconds())/1000)
+	if best {
+		return green + value + reset
+	}
+	if worst {
+		return red + value + reset
+	}
+	return yellow + value + reset
+}
+
+func formatSuccessRate(successful, total int, best bool, worst bool) string {
+	value := fmt.Sprintf("%d/%d", successful, total)
+	if best {
+		return green + value + reset
+	}
+	if worst {
+		return red + value + reset
+	}
+	return yellow + value + reset
+}
+
+func printResults(results []endpointResult, requestCount *int) {
```

Comment:
Can we add a CLI flag to dump the results as e.g. a CSV as well? When we hand this out to ~10 people to collect results it might be nicer when we try to import them into e.g. a google sheet

Reply:
Great idea, added!